Dockerfile

```diff
@@ -0,0 +1,27 @@
+# Use the official OpenJDK base image
+FROM openjdk:17-jdk-buster
+
+# Set the working directory in the container
+WORKDIR /app
+
+# Copy only the Gradle files into the container
+COPY build.gradle.kts settings.gradle.kts /app/
+COPY gradle /app/gradle
+
+# Copy the Gradle Wrapper files
+COPY gradlew gradlew.bat /app/
+
+# Make the Gradle Wrapper scripts executable
+RUN chmod +x /app/gradlew
+
+# Copy the source code into the container
+COPY src /app/src
+
+# Build the Spring Boot application inside the container
+RUN /app/gradlew build
```

Comment:
I don't like you build the inside the container. First of all it's slow. 
- At least you must copy gradle cache inside the docker image to speed up the build
- Second. It would be better to use multistage build feature.

It all about improving your approach. But, let's build the app outside a container and copy assembled jar into the image.

Ask @OlehSarapuk for help :)

Reply:
Thanks, fixed :D